aten/src/ATen/native/ComplexHelper.h

```diff
@@ -13,17 +13,51 @@ inline std::vector<int64_t> computeStrideForComplex(IntArrayRef oldstride) {
   return res;
 }
 
-// expects as input a complex tensor and returns back a float tensor
-// containing the complex values in the last two dimensions
-inline Tensor view_complex_as_float(const Tensor& self) {
-  TORCH_INTERNAL_ASSERT(self.is_complex());
+// expects as input a complex tensor and returns back a tensor
+// with corresponding real dtype containing the complex values
+// in the last two dimensions
+Tensor view_as_real(const Tensor& self) {
+  TORCH_CHECK(self.is_complex(), "view_as_real is only supported for complex tensors");
   auto new_sizes = self.sizes().vec();
   // last dimension will always have two elements containing the real and imag vals
   new_sizes.emplace_back(2);
-  auto new_strides = computeStrideForComplex(self.strides());
+  auto new_strides = computeStrideForViewAsReal(self.strides());
   auto new_storage_offset = 2 * self.storage_offset();
   const auto float_type = c10::toValueType(self.scalar_type());
   return at::empty({0}, self.options().dtype(float_type)).set_(self.storage(), new_storage_offset, new_sizes, new_strides);
 }
 
+inline std::vector<int64_t> computeStrideForViewAsComplex(IntArrayRef oldstride) {
+  auto res = oldstride.vec();
+  int dim = res.size();
+
+  TORCH_CHECK(res[dim-1] == 1, "Tensor must have a last dimension with stride 1");
+  res.pop_back();
+
+  for(size_t i = 0; i < res.size(); i++) {
+    TORCH_CHECK(res[i] % 2 == 0, "Tensor must have a stride divisible by 2 for all but last dimension");
+    res[i] = res[i] / 2;
+  }
+  return res;
+}
+
+// expects as input a float or double tensor with last dimension of size 2
+// and returns back a tensor with corresponding complex dtype
+Tensor view_as_complex(const Tensor& self) {
+  TORCH_CHECK(
+    self.scalar_type() == kFloat || self.scalar_type() == kDouble,
+    "view_as_complex is only supported for float and double tensors");
+
+  auto new_sizes = self.sizes().vec();
+  TORCH_CHECK(new_sizes[self.dim()-1] == 2, "Tensor must have a last dimension of size 2");
+  new_sizes.pop_back();
+
+  auto new_strides = computeStrideForViewAsComplex(self.strides());
```

Comment:
nit: const auto?, since we're labeling things const (which is the right thing to do)